lib/sles4sap/sap_deployment_automation_framework/deployment_connector.pm

```diff
@@ -0,0 +1,233 @@
+
+    return $vm_list[0];
+}
+
+=head2 get_parent_ids
+
+    get_parent_ids();
+
+Returns B<ARRAYREF> of all parent job IDs acquired from current job data.
+
+=cut
+
+sub get_parent_ids {
+    my $job_info = get_job_info(get_current_job_id());
+    # This will loop through all parent job types (chained, parallel, etc...) and creates a list of IDs
+    my @parent_ids = map { @{$job_info->{parents}{$_}} } keys(%{$job_info->{parents}});
+    diag((caller(0))[3] . "Parent job data: " . Dumper($job_info->{parents}));
+    foreach (@parent_ids) { die "Returned parent ID must be a number: '$_'" unless looks_like_number($_); }
+    return \@parent_ids;
+}
+
+=head2 find_deployment_id
+
+    find_deployment_id(deployer_resource_group=>$deployer_resource_group);
+
+B<deployer_resource_group>: Deployer resource group. Default: get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP')
+
+Finds deployment ID for currently running test. Deployment ID is ID of an OpenQA test which created deployer VM.
+In case of multi-machine test this can be either parent test ID or current job id as well.
+This function collects all OpenQA job IDs related to current test run and checks if any of them match an existing
+deployer VM tagged with this ID.
+
+B<Example>:
+Job: 123456 - deployment module - created deployer VM tagged with "deployment_id=123456",
+Job: 123457 (child of 123456) - some test module
+Deployment ID returned from both jobs: 123456 - because it matches with existing VM tagged with "deployment_id=123456"
+
+=cut
+
+sub find_deployment_id {
+    my (%args) = @_;
+    $args{deployer_resource_group} //= get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP');
+    my @check_list = (get_current_job_id(), @{get_parent_ids()});
+
+    diag("Job IDs found: " . Dumper(@check_list));
+    my @ids_found;
+    for my $deployment_id (@check_list) {
+        my $vm_name =
+          get_deployer_vm(deployer_resource_group => $args{deployer_resource_group}, deployment_id => $deployment_id);
+        push(@ids_found, $deployment_id) if $vm_name;
+    }
+    die "More than one deployment found.\nJobs IDs: " .
+      join(', ', @check_list) . "\nVMs found: " . join(', ', @ids_found) if @ids_found > 1;
+
+    return ($ids_found[0]);
+}
+
+=head2 get_deployer_resources
+
+    get_deployer_resources(deployer_resource_group=>$deployer_resource_group, deployment_id=>'123456');
+
+B<deployer_resource_group>: Deployer resource group. Default: get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP')
+
+B<deployment_id>: Deployment ID
+
+B<return_ids>: Returns array of resource IDs instead of names
+
+Returns ARRAYREF of all resources belonging to B<deployer_resource_group> tagged with B<deployment_id>.
+
+=cut
+
+sub find_deployer_resources {
+    my (%args) = @_;
+    $args{deployer_resource_group} //= get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP');
+    $args{deployment_id} //= find_deployment_id();
+    $args{return_ids} = $args{return_ids} ? 'id' : 'name';
+    my $az_cmd = join(' ',
```

Comment:
it is a good candidate for `lib/azure_cli.pm`. Have a look to `az_network_vnet_get`: it return a decoded json, it accept argument about what to query.

Reply:
I am thinking about that ...but on the other side I am being cautious about entangling it too much with other projects. 
So far I would need to import the azure_cli lib only for one function. So Maybe if there are more cases like this.